script.sql

```diff
@@ -0,0 +1,162 @@
+
+
+-- tables
+-- Table: app
+CREATE TABLE app (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(25) NOT NULL,
+    description text NOT NULL,
+    weight int NOT NULL,
+    create_date date NOT NULL,
+    is_free bool NOT NULL,
+    price decimal(8,2) NULL,
+    has_advert bool NOT NULL,
+    category_id int NOT NULL,
+    CONSTRAINT app_pk PRIMARY KEY (id)
+);
+
+-- Table: app_category
+CREATE TABLE app_category (
+    id int NOT NULL AUTO_INCREMENT,
+    name varchar(30) NOT NULL,
+    description text NULL,
+    CONSTRAINT app_category_pk PRIMARY KEY (id)
+);
+
+-- Table: app_creating
+CREATE TABLE app_creating (
+    app_id int NOT NULL,
+    creater_id int NOT NULL,
+    CONSTRAINT app_creating_pk PRIMARY KEY (app_id,creater_id)
+);
+
+-- Table: card
+CREATE TABLE card (
+    number int NOT NULL,
+    bank_name varchar(25) NOT NULL,
+    user_id int NOT NULL,
+    CONSTRAINT card_pk PRIMARY KEY (number)
+);
+
+-- Table: category
+CREATE TABLE category (
+    id int NOT NULL AUTO_INCREMENT,
+    audience_type enum ('child', 'teen', 'mature', 'all') NOT NULL,
+    app_category_id int NOT NULL,
+    CONSTRAINT category_pk PRIMARY KEY (id)
+);
+
+-- Table: creater
+CREATE TABLE creater (
+    id int NOT NULL AUTO_INCREMENT,
+    full_name varchar(60) NOT NULL,
+    work_branch varchar(50) NOT NULL,
+    email varchar(70) NOT NULL,
+    created_app int NULL,
+    CONSTRAINT creater_pk PRIMARY KEY (id)
+);
+
+-- Table: device_type
+CREATE TABLE device_type (
+    device enum('mobile', 'tablet', 'chromebook', 'tv') NOT NULL,
+    app_id int NOT NULL,
+    CONSTRAINT device_type_pk PRIMARY KEY (device)
```

Comment:
Consider adding the first record which would be a mobile game with app_id = 1. Then you add another mobile app with app_id = 37. Since `device` is used as PK here, and two records have the same PK `device="mobile"`, you'll get an error. What do you do?